tasks/iac.yaml

```
@@ -118,18 +120,19 @@ tasks:
               break
             fi
           done
-
           # wait for cluster components
           while true; do
-            if [ $(uds zarf tools kubectl get po,job -A --no-headers=true | egrep -v 'Running|Complete' | wc -l) -gt 0 ]; then
+            if [ $(uds zarf tools kubectl get po,job -A --no-headers=true | egrep -v 'helm-install|Running|Complete' | wc -l) -gt 0 ]; then
```

Probably missing the obvious - what was this change needed for? Iirc this originally was used to make sure all pods/jobs were either completed or running before moving on.

---

This change was added to filter out any of the pods that are spawned by rancher's helm controller. In some cases, a helm release may succeed but the `helm-install` pod dedicated to rolling out the helm release can end up in a failed state.

---

We actually encountered an error that is semi-related over the weekend:
- MetalLB CRDs don't exist: https://github.com/defenseunicorns/uds-core/actions/runs/14013744264/job/39236727846#step:12:39
- Helm install of MetalLB failed/restarting: https://github.com/defenseunicorns/uds-core/actions/runs/14013744264/job/39236727846#step:14:75

While this PR does remove the metallb install I think, I'm worried we might hit similar issues with this check. We probably need one check for `job` resources to make sure they complete, and a separate check for pods to make sure they are running or completed? Do you have more insight into what these "false failures" are? Otherwise I'm worried we'll hit some race conditions here because this will hide real failures.

---

Good point, thanks for calling this out!. I updated the `rke2-cluster-ready` task to wait for all 4 cluster nodes to provision and wait for all `helm-install` pods to be in a `Completed` state before continuing.

---

```suggestion
            if [ $(uds zarf tools kubectl get po,job -A --no-headers=true | egrep -v 'Running|Complete' | wc -l) -gt 0 ]; then
```

If I'm following correctly, helm-installs should already be complete before this, so we don't need to filter them out?